### ISSUE TITLE
Lists mapped to retired Concepticon_IDs

### ISSUE THREAD

**Issue by AnnikaTjuka** [concepticon/concepticon-data#810]
We have a few Concepticon_IDs which are retired but still have lists mapped to them, e.g., 1328 SCISSORS (Replacement_ID: 356 SHEARS). 

Should we move all lists that are mapped to a retired concept to the replacement ID?

**Comment by LinguList** [comment]
Yes, this should be done. In fact, I thought there was even a check that
would prevent this to happen upon PRs. If there is no check, we need
one, @chrzykie, do you know of a test in this direction?

**Comment by chrzyki** [comment]
Heya,

I don't think we explicitly test for that, but I can add it!

This is the list of retired concepts still in use:

```
Savelyev-2020-254-122 uses 2309 which is replaced by 1422
Savelyev-2020-254-176 uses 2275 which is replaced by 1388
Bodt-2019-664-621 uses 1785 which is replaced by 549
Sagart-2019-250-159 uses 1099 which is replaced by 363
Sagart-2019-250-212 uses 2307 which is replaced by 2249
Castro-2010-511-218 uses 2275 which is replaced by 1388
Huang-1992-1820-452 uses 2050 which is replaced by 404
Huang-1992-1820-1452 uses 1099 which is replaced by 363
Huang-1992-1820-1453 uses 1317 which is replaced by 501
Huang-1992-1820-1746 uses 2563 which is replaced by 29
Hale-1973-1798-256 uses 1785 which is replaced by 549
Bowern-2012-644-453 uses 1785 which is replaced by 549
BeijingDaxue-1964-905-527 uses 1317 which is replaced by 501
Chacon-2014-142-85 uses 2138 which is replaced by 2641
List-2014-1280-923 uses 1785 which is replaced by 549
Borin-2015-1532-1038 uses 1785 which is replaced by 549
Key-2016-1310-617 uses 1785 which is replaced by 549
Haspelmath-2009-1460-348 uses 1785 which is replaced by 549
Buck-1949-1110-224 uses 1785 which is replaced by 549
Luniewska-2016-299-116 uses 1328 which is replaced by 356
Luniewska-2019-299-116 uses 1328 which is replaced by 356
Othaniel-2017-300-140 uses 1099 which is replaced by 363
Othaniel-2017-300-205 uses 2275 which is replaced by 1388
Tolmie-1884-211-50 uses 2394 which is replaced by 301
Bremer-2016-200-47 uses 2275 which is replaced by 1388
Snider-2004-1700-175 uses 3163 which is replaced by 296
Snider-2004-1700-487 uses 1785 which is replaced by 549
Snider-2004-1700-587 uses 2050 which is replaced by 404
Snider-2004-1700-966 uses 264 which is replaced by 2475
Snider-2004-1700-1350 uses 2275 which is replaced by 1388
Dawson-1887-731-627 uses 1328 which is replaced by 356
Xiao-2012-213-43 uses 776 which is replaced by 344
Monnier-2014-1031-232 uses 1328 which is replaced by 356
DiezAlamo-2018-750-568 uses 1328 which is replaced by 356
Clark-2008-503-8 uses 2275 which is replaced by 1388
Clark-2008-503-308 uses 1328 which is replaced by 356
List-2018-1105-76 uses 170 which is replaced by 2691
List-2018-1105-85 uses 1984 which is replaced by 2692
List-2018-1105-157 uses 171 which is replaced by 2693
List-2018-1105-596 uses 1785 which is replaced by 549
Rzymski-2020-1624-108 uses 170 which is replaced by 2691
Rzymski-2020-1624-153 uses 1984 which is replaced by 2692
Rzymski-2020-1624-297 uses 171 which is replaced by 2693
Rzymski-2020-1624-906 uses 1785 which is replaced by 549
Rzymski-2020-1624-1584 uses 463 which is replaced by 355
Rzymski-2020-1624-1589 uses 2309 which is replaced by 1422
Rzymski-2020-1624-1620 uses 2490 which is replaced by 12
Rzymski-2020-1624-1621 uses 3173 which is replaced by 3171
Morse-1996-375-163 uses 927 which is replaced by 925
```

quick and dirty with:

```python
from pyconcepticon.api import Concepticon

conc = Concepticon("/PATH/TO/CONCEPTICON")
replacements = {}

for concept_id, conceptset in conc.conceptsets.items(): 
    if conceptset.replacement_id:
        replacements[concept_id] = conceptset.replacement_id

for conceptlist_id, conceptlist in conc.conceptlists.items(): 
    for concept_id, values in conceptlist.concepts.items(): 
        if values.concepticon_id in replacements.keys(): 
            print(f"{values.id} uses {values.concepticon_id} which is replaced by {replacements[values.concepticon_id]}")

**Comment by AnnikaTjuka** [comment]
Thanks for the list @chrzyki ! I'll prepare a PR in which I replace the retired IDs.

**Comment by chrzyki** [comment]
@LinguList do you have any preferences for where this test should happen, i.e. during which stage of contributing? `concepticon test`?

**Comment by LinguList** [comment]
Yes, in the tests. I really think we HAD that test before...

**Comment by AnnikaTjuka** [comment]
> In that case, I would say: we have to write the tests first, before we check the data. So we put this PR on hold until then. And the test should test then:
> 
> 1. is a concepticon ID used that is just an alias for another one?
> 2. are concepticon IDs defined as an alias for another one that should NOT be defined as such, since they contradict the order of IDs (and therefore the history of the project)?

See discussion in PR #816

**Comment by LinguList** [comment]
Update on this issue.

**Comment by LinguList** [comment]
I just made a new check for this problem, and here comes a list of cases where the wrong replacment ID is being used:
``` 
 CONCEPTICON_ID    LINE_NO  ID                           NUMBER  GLOSS
----------------  ---------  -------------------------  --------  ---------------------------
            1317        528  BeijingDaxue-1964-905-527       527  raise (something with hand)
            1785        622  Bodt-2019-664-621       621  mate, have intercourse
            1785       1039  Borin-2015-1532-1038      1038  to have sex
            1785        454  Bowern-2012-644-453       453  sexual intercourse
            2275         48  Bremer-2016-200-47        47  shadow
            1785        225  Buck-1949-1110-224      4.67  coire; copulate; sexual intercourse, have
            2275        219  Castro-2010-511-218       218  shadow
            2138         86  Chacon-2014-142-85        85  mythical being
            2275          9  Clark-2008-503-8           8  shadow
            1328        309  Clark-2008-503-308       308  scissors
            1328        628  Dawson-1887-731-627       627  Scissors
            1328        569  DiezAlamo-2018-750-568       568  scissors
```
One can in all cases just replace the ID and gloss by the replacment ID.

I wrote the test for pyconcepticon. If somebody could then, once this is done, look into this issue, thsi woudl be fine.

BTW: one should always run `$ concepticon check LIST` before submitting, as this reveals more tests than the ones we run in `$concepticon test`, also for uniqueness, etc.






```

**Comment by chrzyki** [comment]
See #899.
